File: packages/db/prisma/seed.ts
import bcrypt from 'bcrypt';
import { PrismaClient, Role } from '@prisma/client';

const prisma = new PrismaClient();

async function ensureUser(email: string, name: string, password: string, role: Role) {
  const passwordHash = await bcrypt.hash(password, 10);
  return prisma.user.upsert({
    where: { email },
    update: { name, passwordHash, role },
    create: { email, name, passwordHash, role }
  });
}

async function main() {
  await ensureUser('operator@has.local', 'Main Operator', 'Operator123!', 'operator');
  await ensureUser('tech@has.local', 'Main Technician', 'Tech123!', 'tech');
  await ensureUser('guest@has.local', 'Guest User', 'Guest123!', 'guest');

  await prisma.tariff.createMany({
    data: [
      { title: 'Домашний 100', speedMbps: 100, price: 600, type: 'home', features: ['Wi-Fi роутер', 'Безлимит'], isFeatured: false, isActive: true },
      { title: 'Турбо 300', speedMbps: 300, price: 900, type: 'home', features: ['Приоритетная поддержка', 'Статический IP'], isFeatured: true, isActive: true },
      { title: 'Бизнес 500', speedMbps: 500, price: 2500, type: 'business', features: ['SLA 99.9%', 'Личный менеджер'], isFeatured: false, isActive: true }
    ],
    skipDuplicates: true
  });

  const now = new Date();
  await prisma.announcement.createMany({
    data: [
      {
        title: 'Плановые работы',
        body: '18 февраля c 02:00 до 04:00 возможны кратковременные перерывы.',
        level: 'warn',
        startsAt: new Date(now.getTime() - 24 * 3600 * 1000),
        endsAt: new Date(now.getTime() + 7 * 24 * 3600 * 1000),
        isActive: true,
        placement: 'top'
      },
      {
        title: 'Подключение за 1 день',
        body: 'Оставьте заявку и подключим интернет в течение 24 часов.',
        level: 'info',
        startsAt: now,
        endsAt: null,
        isActive: true,
        placement: 'homepage'
      }
    ],
    skipDuplicates: true
  });
}

main().finally(async () => prisma.$disconnect());

File: apps/api/core/management/commands/seed.py
from django.core.management.base import BaseCommand
from django.utils import timezone
from datetime import timedelta
from core.models import Announcement, Tariff, User


class Command(BaseCommand):
    help = 'Seed data'

    def handle(self, *args, **options):
        User.objects.update_or_create(
            email='operator@has.local',
            defaults={'name': 'Operator', 'role': 'operator'},
        )
        op = User.objects.get(email='operator@has.local')
        op.set_password('Operator123!')
        op.save()

        User.objects.update_or_create(
            email='tech@has.local',
            defaults={'name': 'Tech', 'role': 'tech'},
        )
        tech = User.objects.get(email='tech@has.local')
        tech.set_password('Tech123!')
        tech.save()

        for i in range(1, 6):
            Tariff.objects.update_or_create(
                title=f'Tariff {i}',
                defaults={
                    'speedMbps': i * 100,
                    'price': i * 500,
                    'type': 'home' if i < 4 else 'business',
                    'features': [f'Feature {i}', '24/7 support'],
                    'isFeatured': i == 3,
                    'isActive': True,
                },
            )

        now = timezone.now()
        for i, level in enumerate(['info', 'warn', 'critical'], 1):
            Announcement.objects.update_or_create(
                title=f'Announcement {i}',
                defaults={
                    'body': f'Important message {i}',
                    'level': level,
                    'startsAt': now - timedelta(days=1),
                    'endsAt': now + timedelta(days=7),
                    'isActive': True,
                    'placement': 'homepage',
                },
            )
        self.stdout.write(self.style.SUCCESS('Seed completed'))
